src/processes/delete-expired-post-process.js

```diff
@@ -24,14 +24,15 @@ const deleteExpiredPostProcess = async (job, done, queueInjection) => {
             expired_posts[doc.author_id] = [doc._id];
           }
         });
+        // eslint-disable-next-line eqeqeq
         if (expired_posts != {}) {
           // Remove from getstream
           for (const [key, value] of Object.entries(expired_posts)) {
             value.forEach(async (act_id) => {
               queueInjection?.add({
-                feedName: 'user_excl',
-                userId: key,
-                activityId: act_id
+                feed_group: 'user_excl',
+                feed_id: key,
+                activity_id: act_id
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block contains an asynchronous operation within a forEach loop. This can lead to unexpected behavior as forEach will not wait for promises to resolve before moving on to the next iteration. Consider using a for...of loop or map with Promise.all instead.
```diff
-              value.forEach(async (act_id) => {
-                queueInjection?.add({
-                 feed_group: 'user_excl',
-                 feed_id: key,
-                 activity_id: act_id
-               });
-               await db.collection(DB_COLLECTION_POST_SCORE).deleteOne({_id: act_id});
-             });
+              for (const act_id of value) {
+                queueInjection?.add({
+                  feed_group: 'user_excl',
+                  feed_id: key,
+                  activity_id: act_id
+                });
+                await db.collection(DB_COLLECTION_POST_SCORE).deleteOne({_id: act_id});
+              }
```
Please note that this change assumes that the order of processing these actions doesn't matter. If the order does matter, you would need to use a for...of loop without Promise.all.


<!-- This is an auto-generated comment by OSS CodeRabbit -->